### PR TITLE
Fix for WorldToScreen with zoomed scopes

### PR DESCRIPTION
**Pull request by Outshynd**
This PR implements optic zoom WorldToScreen resolution for both pings and coop player nameplates/health. Tested with Vudu 1x/6x, Valday 1x/6x, ADO 3x/9x, and Specter 1x/4x optics. Does not change default behavior for any reflex sight or optic with zoom level <= 1.0. Did not test with SSAA, but it shouldn't affect anything as both the worldCamera and opticCamera should scale the same and the screen scale logic was not changed.

### CODE REVIEW

File: Fika.Core/Coop/Custom/FikaHealthBar.cs [hunk 1]
```diff
@@ -98,9 +99,8 @@ private void UpdateScreenSpacePosition(bool throttleUpdate)
 
             float processedDistance = Mathf.Clamp(sqrDistance / 625, 0.6f, 1f);
             Vector3 position = new(currentPlayer.PlayerBones.Neck.position.x, currentPlayer.PlayerBones.Neck.position.y + (1f * processedDistance), currentPlayer.PlayerBones.Neck.position.z);
-            Vector3 screenPoint = camera.WorldToScreenPoint(position);
-
-            if (screenPoint.z <= 0)
+            
+            if (WorldToScreen.GetScreenPoint(position, mainPlayer, out Vector3 screenPoint) == false)
```
Comment (Lacyway) [inline Fika.Core/Coop/Custom/FikaHealthBar.cs:103]: Could we change this to:
```cs
if (!WorldToScreen.GetScreenPoint(position, mainPlayer, out Vector3 screenPoint)
```

File: Fika.Core/Coop/Factories/PingFactory.cs [hunk 2]
```diff
@@ -75,19 +76,16 @@ protected void Update()
...
                 float inputWidth = CameraClass.Instance.SSAA.GetInputWidth();
                 screenScale = outputWidth / inputWidth;
             }
 
-            Vector3 screenPoint = camera.WorldToScreenPoint(hitPoint);
-            if (screenPoint.z > 0)
+            if (WorldToScreen.GetScreenPoint(hitPoint, mainPlayer, out Vector3 screenPoint))
             {
-                float distanceToCenter = Vector3.Distance(screenPoint, new Vector3(x: Screen.width, Screen.height, 0) / 2);
+                float distanceToCenter = Vector3.Distance(screenPoint, new Vector3(x: Screen.width, y: Screen.height, 0) / 2);
```
Comment (Lacyway) [inline Fika.Core/Coop/Factories/PingFactory.cs:88]: We could remove `x:` and `y:` here, not sure why they were the originally. Must've been a mistake on my end.

File: Fika.Core/Utils/WorldToScreen.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    public static class WorldToScreen
+    {
+        public static bool GetScreenPoint(Vector3 worldPosition, CoopPlayer mainPlayer, out Vector3 screenPoint)
+        {
+            CameraClass worldCameraInstance = CameraClass.Instance;
+            Camera worldCamera = worldCameraInstance.Camera;
+
+            screenPoint = Vector3.zero;
+
+            if (mainPlayer == null || worldCamera == null)
```
Comment (Lacyway) [inline Fika.Core/Utils/WorldToScreen.cs:18]: Could we change to, and in other places as well
```cs
if (mainPlayer == null || worldCamera == null)
{
    return false;
}
```
Just to keep scopes consistent throughout the repo, I've been cleaning up old code recently to achieve the same.

File: Fika.Core/Coop/Factories/PingFactory.cs (hunk 2 above)
Comment (Lacyway) [inline Fika.Core/Coop/Factories/PingFactory.cs:88]: We could remove `x:` and `y:`, they must've been left there mistakenly in the old code.

**Review by Lacyway — CHANGES_REQUESTED** [review]
Tested locally with a PiP scope, works great.
Thank you so much. Mind looking into the small details I left? Thanks!

File: Fika.Core/Coop/Factories/PingFactory.cs (hunk 2 above)
Comment (Outshynd) [reply Fika.Core/Coop/Factories/PingFactory.cs]: Done

Comment (Lacyway) [reply Fika.Core/Coop/Factories/PingFactory.cs]: You're awesome! Thanks.

**Pull request merged**